### PR TITLE
Open bsd power pc

### PR DESCRIPTION
**Pull request by seanm**

### CODE REVIEW

**Comment by seanm** [comment]
@malaterre please check carefully. I don't know this code at all.

Also, I'm quite confused by some of this swapping code... I'll add some questions in-line.

File: Source/Common/gdcmByteSwap.txx [hunk 1]
```diff
@@ -106,7 +106,7 @@ void Swap4(T &a, SwapCode const &swapcode)
 {
```
Comment (seanm) [inline Source/Common/gdcmByteSwap.txx:106]: When is `Swap4` meant to swap?  It seems to swap when the given `swapcode` *is opposite* the host endianness.

**Comment by malaterre** [comment]
@seanm could you please rebase your branch and push again. Thanks much!

**Comment by seanm** [comment]
> @seanm could you please rebase your branch and push again. Thanks much!

Done.

**Comment by malaterre** [comment]
I've integrated most of these changes. Feel free to open a new one, if you find remaining issues. Thanks much sean !

File: Source/Common/gdcmByteSwap.txx (hunk 1 above)
Comment (seanm) [reply Source/Common/gdcmByteSwap.txx:106]: @malaterre do you have an answer for this?

File: Source/Common/gdcmByteSwap.txx [hunk 2]
```diff
@@ -106,7 +106,7 @@ void Swap4(T &a, SwapCode const &swapcode)
...
 #ifndef GDCM_WORDS_BIGENDIAN
   if ( swapcode == 4321 || swapcode == 2143 )
-    a = (T)(( a << 8 ) | ( a >> 8 ));
+    a = ( a << 8 ) | ( a >> 8 );
```
Comment (seanm) [inline Source/Common/gdcmByteSwap.txx:109]: @malaterre I see you didn't take this change. It was meant to make line 109 the same as line 112. I don't see why these two lines should be different...

File: Source/Common/gdcmSwapper.txx [hunk 3]
```diff
@@ -127,12 +127,12 @@ namespace gdcm
 
   template <> inline Tag SwapperNoOp::Swap<Tag>(Tag val)
     {
-    return Tag( Swap(val.GetGroup()), Swap(val.GetElement()) );
+    return Tag( Swap((uint32_t)val.GetElementTag()) );
```
Comment (seanm) [inline Source/Common/gdcmSwapper.txx:130]: @malaterre this change I don't know what it does, but it too was to make line 130 the same as line 200. Surely they should be the same?

**Pull request closed without merging**